libs/@guardian/libs/src/performance/startPerformanceMeasure.ts

```diff
@@ -23,29 +28,35 @@ export const startPerformanceMeasure = (
 	name: string,
 	action?: string,
 ): { endPerformanceMeasure: () => number } => {
-	if (!('getEntriesByName' in window.performance)) {
-		return fallback;
-	}
+	try {
+		const measureName = serialise({ team, name, action });
+		const markName = `${measureName}-${getId()}`;
+		const start = performance.now();
 
-	const measureName = serialise(team, name, action);
+		// Browser support for `measureOptions` is not good enough,
+		// so we have to rely on the side effect of adding a mark.
+		// See https://developer.mozilla.org/en-US/docs/Web/API/Performance/measure#measureoptions
+		performance.mark(markName);
 
-	const options = {
-		start: performance.now(),
-		detail: {
-			team,
-			name,
-			action,
-		},
-	} satisfies PerformanceMeasureOptions;
+		const endPerformanceMeasure = () => {
+			try {
+				const { duration } = window.performance.measure(
+					measureName,
+					markName,
+				) ?? {
+					duration: performance.now() - start,
+				};
 
-	const endPerformanceMeasure = () => {
-		const { duration } = window.performance.measure(measureName, options) ??
-			window.performance.getEntriesByName(measureName, 'measure').at(-1) ?? {
-				duration: performance.now() - options.start,
-			};
+				return Math.ceil(duration);
+			} catch (error) {
+				log(team, error);
+				return fallbackDuration;
+			}
+		};
 
-		return Math.ceil(duration);
-	};
-
-	return { endPerformanceMeasure };
+		return { endPerformanceMeasure };
+	} catch (error) {
+		log(team, error);
+		return { endPerformanceMeasure: () => fallbackDuration };
```

Comment:
If we use this to analyse data, would `-1` skew that?

Reply:
It could, but we're not directly using it yet. I followed the pattern of [`indexOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf), [`findIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) and the likes returning **-1**, which may have been ill-advised.

Would `undefined` be more accurate? It would require a major version bump.